tests/ResponseDocumentingTest.php

```diff
@@ -96,3 +98,57 @@ public function index()
         return response()->json(['foo' => 'bar']);
     }
 }
+
+test('laravel responder success support', function () {
+    \Illuminate\Support\Facades\Route::get('api/test', [Foo_TestFive::class, 'index']);
+
+    \Dedoc\Scramble\Scramble::routes(fn (Route $r) => $r->uri === 'api/test');
+    $openApiDocument = app()->make(\Dedoc\Scramble\Generator::class)();
+
+    assertMatchesSnapshot($openApiDocument);
+});
+class Foo_TestFive
+{
+    public function index()
+    {
+        $user = [];
+        return responder()->success($user, UserTransformer::class)->respond(201, [
+            'Authorization' => 'Bearer ghjkiuygvbnmkuyg'
+        ]);
+    }
+}
+
+/**
+ * @mixin User
+ */
+class UserTransformer extends TransformerAbstract
+{
+    public function transform(User $user): array
+    {
+        return [
+            'id' => $user->id,
+            'name' => $user->name,
+            'testString' => 'iasdihas',
+            'testNumber' => 2333
+        ];
+    }
+}
+
+test('laravel responder error support', function () {
+    \Illuminate\Support\Facades\Route::get('api/test', [Foo_TestFive::class, 'index']);
+
+    \Dedoc\Scramble\Scramble::routes(fn (Route $r) => $r->uri === 'api/test');
+    $openApiDocument = app()->make(\Dedoc\Scramble\Generator::class)();
+
+    assertMatchesSnapshot($openApiDocument);
+});
+class Foo_TestSix
+{
+    public function index()
+    {
+        $user = [];
+        return responder()->error($user, UserTransformer::class)->respond(404, [
```

Comment:
This wont work ATM, the library checks if the first param of respond is of type number, so it will probably crash or avoid the route if you put a value from an enum, even though the underlying value is an int

Reply:
Actually, `Response` is a class, not an enum, and `HTTP_NOT_FOUND` is a constant which equals to a number (`int`), which means it should in fact work. Lets use the `Response` class constants for HTTP statuses to avoid magic numbers.

Reply:
I agree it should work, it's not working in this iteration, since it's the same reasoning as with an enum. The type of HTTP_NOT_FOUND is first a Class constant in a syntax tree, even though, at the end the underlying type is an int

Reply:
Oh okay. I explained it because I thought you thought `Response` is an enum. It's not as simple as I believed it seems. Thanks for the concise explanation.